src/test/java/be/seeseemelk/mockbukkit/entity/FishHookMockTest.java

```diff
@@ -0,0 +1,177 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.MockBukkit;
+import be.seeseemelk.mockbukkit.ServerMock;
+import be.seeseemelk.mockbukkit.WorldMock;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.FishHook;
+import org.bukkit.entity.Player;
+import org.bukkit.entity.SpawnCategory;
+import org.bukkit.util.Vector;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class FishHookMockTest
+{
+
+	private ServerMock server;
+	private WorldMock world;
+	private FishHookMock hook;
+
+	@BeforeEach
+	public void setUp()
+	{
+		server = MockBukkit.mock();
+		world = server.addSimpleWorld("world");
+		hook = new FishHookMock(server, UUID.randomUUID());
+	}
+
+	@AfterEach
+	public void tearDown()
+	{
+		MockBukkit.unmock();
+	}
+
+	@Test
+	void setMinWaitTime_Valid()
+	{
+		FishHookMock hook = new FishHookMock(server, UUID.randomUUID());
+		hook.setMinWaitTime(1);
+		assertEquals(1, hook.getMinWaitTime());
+	}
+
+	@Test
+	void setMaxWaitTime_Valid()
+	{
+		FishHookMock hook = new FishHookMock(server, UUID.randomUUID());
+		hook.setMaxWaitTime(1_000);
+		assertEquals(1_000, hook.getMaxWaitTime());
+	}
+
+	@Test
+	void setWaitTime_Invalid()
+	{
+		assertThrows(IllegalArgumentException.class, () ->
+		{
+			hook.setMaxWaitTime(5);
+			hook.setMinWaitTime(10);
+		});
+	}
+
+	@Test
+	void applyLure()
+	{
+		hook.setApplyLure(false);
+		assertFalse(hook.getApplyLure());
+		hook.setApplyLure(true);
+		assertTrue(hook.getApplyLure());
+	}
+
+	@Test
+	void setBiteChance_Valid()
+	{
+		hook.setBiteChance(0.5);
+		assertEquals(0.5, hook.getBiteChance());
+	}
+
+	@Test
+	void setBiteChance_Invalid()
+	{
+		assertThrows(IllegalArgumentException.class, () ->
+		{
+			hook.setBiteChance(2);
+		});
+		assertThrows(IllegalArgumentException.class, () ->
+		{
+			hook.setBiteChance(-1);
+		});
+	}
+
+// todo: Uncomment when WorkMock#isClearWeather is implemented.
+//	@Test
+//	void setBiteChance_Raining()
+//	{
+//		hook.setBiteChance(-1);
+//		assertEquals(1.0 / 500.0, hook.getBiteChance());
+//		world.setThunderDuration(60);
+//		assertEquals(1.0 / 300.0, hook.getBiteChance());
+//	}
```

Comment:
Let's implement that feature first so we can make this PR great again

Reply:
Implemented in #379 so just waiting on that.